src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexMatcher.cs

```diff
@@ -374,260 +363,207 @@ public SymbolicMatch FindMatch(bool isMatch, ReadOnlySpan<char> input, int start
             }
 
             // Phase 3:
-            // Match again, this time from the computed start position, to find the latest end position.  That start
-            // and end then represent the bounds of the match.  If the pattern has subcaptures (captures other than
-            // the top-level capture for the whole match), we need to do more work to compute their exact bounds, so we
-            // take a faster path if captures aren't required.  Further, if captures aren't needed, and if any possible
-            // match of the whole pattern is a fixed length, we can skip this phase as well, just using that fixed-length
-            // to compute the ending position based on the starting position.  Continuing the previous example, phase 3
-            // will walk forwards from the 6th a until it finds the end of the match: aaaaaaaaaabbb.
+            // If there are no subcaptures, the matching process is done.  For patterns with subcaptures (captures other
+            // than the top-level capture for the whole match), we need to do an additional pass to find their bounds.
+            // Continuing for the previous example, phase 3 will be executed for the characters inside the match, aaabbbc,
+            // and will find associate the one capture (b*) with it's match: bbb.
             if (!HasSubcaptures)
             {
-                if (_fixedMatchLength.HasValue)
-                {
-                    return new SymbolicMatch(matchStart, _fixedMatchLength.GetValueOrDefault());
-                }
-
-                int matchEnd = FindEndPosition(input, matchStart, perThreadData);
-                return new SymbolicMatch(matchStart, matchEnd + 1 - matchStart);
+                return new SymbolicMatch(matchStart, matchEnd - matchStart);
             }
             else
             {
-                int matchEnd = FindEndPositionCapturing(input, matchStart, out Registers endRegisters, perThreadData);
-                return new SymbolicMatch(matchStart, matchEnd + 1 - matchStart, endRegisters.CaptureStarts, endRegisters.CaptureEnds);
+                Registers endRegisters = FindSubcaptures(input, matchStart, matchEnd, perThreadData);
+                return new SymbolicMatch(matchStart, matchEnd - matchStart, endRegisters.CaptureStarts, endRegisters.CaptureEnds);
             }
         }
 
-        /// <summary>Phase 3 of matching. From a found starting position, find the ending position of the match using the original pattern.</summary>
-        /// <remarks>
-        /// The ending position is known to exist; this function just needs to determine exactly what it is.
-        /// We need to find the longest possible match and thus the latest valid ending position.
-        /// </remarks>
+        /// <summary>Performs the initial Phase 1 match to find the end position of the match, or first final state if this is an isMatch call.</summary>
         /// <param name="input">The input text.</param>
```

Comment:
Thank you for reordering these methods to match how they're used.  It confused me every time I read through this file :)